packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -112,6 +112,11 @@ export interface IFunction extends IResource, ec2.IConnectable, iam.IGrantable {
    * Configures options for asynchronous invocation.
    */
   configureAsyncInvoke(options: EventInvokeConfigOptions): void;
+
+  /**
+   * Checks whether this function is compatible for Lambda@Edge.
+   */
+  checkEdgeCompatibility(): void;
```

Comment:
Why have you added this API to `IFunction`? Is there value in exposing this to users?

Reply:
Because the underlying Lambda of a version is an `IFunction`. For the same reason, I had to make `checkEdgeCompatibility()` public. Do you see a workaround?

Reply:
Sorry, I was not able to follow your reasoning. Can you expand a bit more?

If we wanted it to be on a base class so that it's available on all implementations, we could make it a protected abstract method on `FunctionBase` and mark it as `@internal`?
At the calling point check on `instanceof` and call this internal method. We do that in a few places in the CDK, like [here](https://github.com/aws/aws-cdk/blob/c570d9cbd0ec618c84f5ac5c2e3256f3d3671a20/packages/%40aws-cdk/aws-apigateway/lib/method.ts#L218).

Reply:
OK with `instanceof` I can move this method to `Function` but it will remain public since I need to call it from a `Version` on its underlying Lambda?

Reply:
If it's moved to `FunctionBase` and marked `@internal`, keeping it as `public` is fine.

Reply:
Also do you prefer the `Symbol` implementation?

Reply:
That's my preference but not strongly.
The value of the symbol based implementation will diminish over the next few months with our upcoming move to monolithic packaging.

Reply:
> That's my preference but not strongly.

Went for the `instanceof` implementation because `FunctionBase` has currently neither a `constructor` nor `FunctionBaseProps`.